provider/core/src/resource.rs

```diff
@@ -79,10 +80,64 @@ impl writeable::Writeable for ResourceCategory {
 ///
 /// Use [`resource_key!`] as a shortcut to create resource keys in code.
 #[derive(PartialEq, Eq, PartialOrd, Ord, Copy, Clone)]
+#[repr(C)]
 pub struct ResourceKey {
+    _tag0: TinyStr8,
```

Comment:
thought: not a huge fan of this, but I guess it's no big deal and it does seem to be the best way to achieve this